service/matching/handler/engine.go

```diff
@@ -677,7 +678,8 @@ pollLoop:
 		task, err := e.getTask(pollerCtx, taskListID, maxDispatch, taskListKind)
 		if err != nil {
 			// TODO: Is empty poll the best reply for errPumpClosed?
-			if errors.Is(err, tasklist.ErrNoTasks) || errors.Is(err, errPumpClosed) {
+			var taskListNotOwnedByHostError *cadence_errors.TaskListNotOwnedByHostError
+			if errors.Is(err, tasklist.ErrNoTasks) || errors.Is(err, errPumpClosed) || errors.As(err, &taskListNotOwnedByHostError) {
```

Comment:
The same here.

Reply:
I would suggest not returning an emptyPollForActivityResponse here since that ... might cause weird things with the poll autoscaler behaviour causing it to backoff polling. I would be preferable to treat this as an error that is retriable, but only at the layer which handles host lookup, and so therefore breaking these lower-retry loops is good, and that should force Frontend to retry

Reply:
Reverted this change - we just return the error and retry on frontend